gem/gem.py

```diff
@@ -621,6 +647,31 @@ def get_hash(self):
         return hash((type(self), self.shape, self.children))
 
 
+class Delta(Scalar, Terminal):
+    __slots__ = ('i', 'j')
+    __front__ = ('i', 'j')
+
+    def __new__(cls, i, j):
+        assert isinstance(i, IndexBase)
+        assert isinstance(j, IndexBase)
+
+        # \delta_{i,i} = 1
+        if i == j:
+            return one
+
+        # Fixed indices
+        if isinstance(i, int) and isinstance(j, int):
+            return Literal(int(i == j))
+
+        self = super(Delta, cls).__new__(cls)
+        self.i = i
+        self.j = j
+        # Set up free indices
+        free_indices = tuple(index for index in (i, j) if isinstance(index, Index))
+        self.free_indices = tuple(unique(free_indices))
```

Comment:
Oh, you care about ordering, making `Delta(j, i)` and `Delta(i, j)` have the same ordering of `free_indices` by calling unique here?